task/03-date-tasks.js

```diff
@@ -76,9 +85,25 @@ function isLeapYear(date) {
  *    Date(2000,1,1,10,0,0),  Date(2000,1,1,15,20,10,453)   => "05:20:10.453"
  */
 function timeSpanToString(startDate, endDate) {
-   throw new Error('Not implemented');
-}
+   const startDate1 = new Date(startDate);
+   const endDate1 = new Date(endDate);
+
+   let timestamp = endDate1.getTime()-startDate1.getTime();
 
+   let milliseconds = parseInt(timestamp % 1000);
+   let seconds = parseInt((timestamp / 1000) % 60);
+   let minutes = parseInt((timestamp / (1000 * 60)) % 60);
+   let hours = parseInt((timestamp / (1000 * 60 * 60)));
+    
+   hours = (hours < 10) ? "0" + hours : hours;
+   minutes = (minutes < 10) ? "0" + minutes : minutes;
+   seconds = (seconds < 10) ? "0" + seconds : seconds;
+   milliseconds = (milliseconds < 10) ? "00" + milliseconds
+      : (milliseconds < 100) ? "0" + milliseconds
+      : milliseconds;
```

Comment:
Please, avoid this code style, if-else is more readable than multiple conditional operators ? :